crates/relayer-framework/src/tests/relayer_mock/base/types/runtime.rs

```diff
@@ -0,0 +1,51 @@
+use alloc::boxed::Box;
+use async_trait::async_trait;
+use core::{future::Future, time::Duration};
+use std::time::Instant;
+use tokio::time::sleep;
+
+use ibc_relayer_runtime::tokio::context::TokioRuntimeContext;
+
+use crate::{
+    base::one_for_all::traits::runtime::{LogLevel, OfaRuntime},
+    tests::relayer_mock::base::error::Error,
+};
+
+pub type MockRuntimeContext = TokioRuntimeContext<Error>;
+
+#[async_trait]
+impl OfaRuntime for MockRuntimeContext {
+    type Error = Error;
+
+    type Time = Instant;
+
+    async fn log(&self, level: LogLevel, message: &str) {
+        match level {
+            LogLevel::Error => tracing::error!(message),
+            LogLevel::Warn => tracing::warn!(message),
+            LogLevel::Info => tracing::info!(message),
+            LogLevel::Debug => tracing::debug!(message),
+            LogLevel::Trace => tracing::trace!(message),
+        }
+    }
+
+    async fn sleep(&self, duration: Duration) {
+        sleep(duration).await;
+    }
+
+    fn now(&self) -> Instant {
+        Instant::now()
```

Comment:
The time itself should be mocked. You can have an internal `MockTimestamp` type that is a newtype over `u128` as seconds or milliseconds.

When `now()` is called, the mock runtime can return the current timestamp, and then increment the internal timestamp by one so that the next call will return a slightly later time.